Megatron Export Update

Davood-M opened the pull request:
Signed-off-by: David Mosallanezhad <dmosallanezh@nvidia.com>

# What does this PR do ?

Updates export process for Megatron-based models

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Changed ORT optimization due to a bug in onnxruntime (https://github.com/microsoft/onnxruntime/issues/12579)

# Usage

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [X] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [X] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.


## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

borisfom commented on nemo/utils/export_utils.py:
```diff
@@ -263,7 +282,7 @@ def replace_FusedLayerNorm(n: nn.Module) -> Optional[nn.BatchNorm2d]:
             mod = nn.LayerNorm(n.normalized_shape, eps=n.eps, elementwise_affine=n.elementwise_affine,).to(dev)
         elif isinstance(n, FastLayerNorm):
-            mod = nn.LayerNorm(n.weight.shape, eps=n.epsilon, elementwise_affine=True, dtype=torch.float16,).to(dev)
+            mod = nn.LayerNorm(n.weight.shape, eps=n.epsilon, elementwise_affine=True, dtype=torch.float,).to(dev)
```
Should we keep original dtype here ?

borisfom replied:
Also, as part of general export process, we currently guard LayerNorm with fp32 casts. So this won't have any effect. If in fact protecting LayerNorm from autocast is not necessary - we should revise that part as well.

borisfom reviewed (changes requested):
Looks good overall. We should resolve LayerNorm question - if it does not have to be guarded, we should remove CastToFloat wrapping pass.

Davood-M commented:
Thanks for the review! I've updated the LayerNorm and added autocast for the following issue that affected newer Megatron models:

`RuntimeError: expected scalar type Half but found Float`
